test/cases/lib/providers/StateProvider.json

```diff
@@ -0,0 +1,345 @@
+			],
+			"uri": "/state/val1/val2?a=val3&b=val4",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1"
+				},
+				"Store2": {
+					"arg1": "val1",
+					"arg2": "val2"
+				},
+				"Store3": {
+					"arg4": "val4"
+				}
+			}
+		},
+		{
+			"name": "should support another order of query parameters",
+			"routes": [
+				"/state/:arg1[Store1, Store2]/:arg2[Store2]?a=:arg3[Store1]&b=:arg4[Store3]"
+			],
+			"uri": "/state/val1/val2?b=val4&a=val3",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1",
+					"arg3": "val3"
+				},
+				"Store2": {
+					"arg1": "val1",
+					"arg2": "val2"
+				},
+				"Store3": {
+					"arg4": "val4"
+				}
+			}
+		},
+		{
+			"name": "should support when query parameters are empty",
+			"routes": [
+				"/state/:arg1[Store1, Store2]/:arg2[Store2]?a=:arg3[Store1]&b=:arg4[Store3]"
+			],
+			"uri": "/state/val1/val2?",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1"
+				},
+				"Store2": {
+					"arg1": "val1",
+					"arg2": "val2"
+				}
+			}
+		},
+		{
+			"name": "should support when query parameter is empty",
+			"routes": [
+				"/state/:arg1[Store1, Store2]/:arg2[Store2]?a=:arg3[Store1]&b=:arg4[Store3]"
+			],
+			"uri": "/state/val1/val2?a=&b=val4",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1",
+					"arg3": ""
+				},
+				"Store2": {
+					"arg1": "val1",
+					"arg2": "val2"
+				},
+				"Store3": {
+					"arg4": "val4"
+				}
+			}
+		},
+		{
+			"name": "should support when static path without parameters",
+			"routes": [
+				"/state?a=:arg3[Store1]&b=:arg4[Store3]"
+			],
+			"uri": "/state?a=val3&b=val4",
+			"expectedState": {
+				"Store1": {
+					"arg3": "val3"
+				},
+				"Store3": {
+					"arg4": "val4"
+				}
+			}
+		},
+		{
+			"name": "should support overriding path parameters by query string parameters",
+			"routes": [
+				"/state/:arg1[Store1, Store2]/:arg3[Store1]?a=:arg3[Store1]&b=:arg4[Store3]"
+			],
+			"uri": "/state/val1/val2?a=valX&b=val4",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1",
+					"arg3": "valX"
+				},
+				"Store2": {
+					"arg1": "val1"
+				},
+				"Store3": {
+					"arg4": "val4"
+				}
+			}
+		},
+		{
+			"name": "should support partial parameters",
+			"routes": [
+				"/state/w:arg1[Store1, Store2]q/q:arg2[Store2]s?a=a:arg3[Store1]z&b=x:arg4[Store3]y"
+			],
+			"uri": "/state/wval1q/qval2s?a=aval3z&b=xval4y",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1",
+					"arg3": "val3"
+				},
+				"Store2": {
+					"arg1": "val1",
+					"arg2": "val2"
+				},
+				"Store3": {
+					"arg4": "val4"
+				}
+			}
+		},
+		{
+			"name": "should support partial parameters and set them empty if they are absent in query",
+			"routes": [
+				"/state/w:arg1[Store1, Store2]q/q:arg2[Store2]s?a=a:arg3[Store1]z&b=x:arg4[Store3]y"
+			],
+			"uri": "/state/wval1q/qval2s?a=val3&b=val4",
+			"expectedState": {
+				"Store1": {
+					"arg1": "val1",
+					"arg3": ""
```

Comment:
Because the parameter exists but its value doesn't match to the pattern. In my opinion, this is an expected behavior.